src/screens/post/components/categoryItem.js

```diff
@@ -0,0 +1,67 @@
+import {View, Text, TouchableOpacity, StyleSheet} from "react-native"
+import React from "react"
+import {useTheme} from "@react-navigation/native"
+import fonts from "@assets/fonts"
+
+const CategoryItem = ({
+  item,
+  selectCategoryId,
+  setSelectCategoryId,
+  setNews,
+  setDuplicateNews,
+  setLastDocument,
+}) => {
+  const {colors} = useTheme()
+  const styles = makeStyles(colors)
+
+  const onToggleCategory = () => {
+    setSelectCategoryId(item.id)
+    setNews([])
+    setDuplicateNews([])
+    setLastDocument()
+  }
```

Comment:
Should pass a function onToggleCategory to this component instead of pass all set state function and call it inside this component